### DISCUSSION TITLE
Setter Injection?

### DISCUSSION THREAD

**Discussion by denchic45** [evant/kotlin-inject#229] [Q&A] [answered]
Is it possible to inject dependencies into fields and not through the constructor? For example, in the same activity or fragments without factories, or is this option not supported?

**Comment by evant** [comment, marked as answer]
This is intentionally not supported because it allows objects to be in a partially constructed state. I have some guidance for android specifically since it does use the pattern where you don't control construction a lot which is unfortunate. https://github.com/evant/kotlin-inject/blob/main/docs/android.md
